app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,67 @@
+class TeachersController < ApplicationController
+  before_action :authenticate_user!, :set_user
+
+  def show
+    @groups = @user.groups.all
+    @total_of_students = @user.group_members.count
+    # TODO: Add days of the weey to this query
+    @in_progress_group = @user.groups
+                              .where("from_time <= ?", Time.new)
+                              .where("to_time >= ?", Time.new).last
+  end
+
+  def show_students
+    @students = User.find(
+      [@user.group_members.map(&:user_id)]
+    )
+    render "teachers/students/index"
+  end
+
+  def new
+    @new_user = User.new
+  end
+
+  def edit; end
+
+  def create
+    @new_user = User.new(user_params)
+    # TODO: Add support to create students user
+    temp_password = @new_user.birth_date.strftime('%d%m%y')
+    @new_user.password = temp_password
+
+    if @new_user.save
+      redirect_to teacher_path(@user),
+                  notice: "#{@new_user.role.capitalize} was successfully created with password #{temp_password}."
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @user.update(user_params)
+      redirect_to teacher_url(@user), notice: "#{@user.role.capitalize} was successfully updated."
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user.destroy
+    redirect_to teachers_url, notice: "#{@user.role.capitalize} was successfully destroyed."
+  end
+
+  private
+
+  def set_user
+    user_id = if params.key?("id")
```

Comment:
try using something like this to keep the code cleaner

```suggestion
    user_id = current_user.id || params[:id]
```